packages/shared/src/react/hooks/usePagesOrInfinite.ts

```diff
@@ -93,22 +156,26 @@ export const usePagesOrInfinite: UsePagesOrInfinite = (params, fetcher, config,
     pageSize: pageSizeRef.current,
   };
 
+  // cacheMode being `true` indicates that the cache key is defined, but the fetcher is not.
+  // This allows to ready the cache instead of firing a request.
+  const shouldFetch = !triggerInfinite && enabled && (!cacheMode ? !!fetcher : true);
+  const swrKey = shouldFetch ? pagesCacheKey : null;
+  const swrFetcher =
+    !cacheMode && !!fetcher
+      ? (cacheKeyParams: Record<string, unknown>) => {
+          // @ts-ignore
```

Comment:
can we document the ts-ignore here?

Reply:
[nit] I would also say that it would be preferable to use `@ts-expect-error` since those have to get cleaned up when the underlying TS issue is resolved